Tutorial.md

```diff
@@ -0,0 +1,121 @@
+# Tutorial on Show Design
+
+## Overview
+
+Sparkle Motion is a tool to design and run a light show.
+
+First, you define a "scene" -- an arrangement of physical light fixtures and their controller mappings.
+
+Next, you design a "show" -- a set of visual effects to project on a scene.
+
+Visual effects programmed as "shaders" -- small GLSL programs that process inputs like xy coordinates from the scene, colors from another shader, and audio metadata from the sound analysis plugin, and generate outputs like another xy coordinate for distortion or projection mapping, or a color for a light.
+
+Finally, you use the designer mode to design a UI for all the "controls" and run the light show in real time.
+
+## Simulator Quick Start
+
+On OS X, start the simulator:
+
+```
+brew install openjdk
+export PATH="/opt/homebrew/opt/openjdk/bin:$PATH"
+./gradlew --continuous jsRun
+```
+
+A brand new session starts with no Show or Scene loaded, and the Show/Scene menu open. For an existing session click the "☰" icon to open the Show/Scene menu.
+
+## Scene Quick Start
+
+Load a scene from the BAAAHS template. Defining a scene's model, light controllers and fixtures is an exercise for another day...
+
+### Load the BAAAHS template
+
+- "☰" to open the Show/Scene menu then Scene tab
+- "+ New Scene"
+- From template "BAAAHS" to see the sheep model 🐑
+- "Save" button, enter "my-baaahs", then "save"
+
+## Show Quick Start
+
+Next we'll create an empty show, define a default projection, then program our first effect.
+
+### Create an empty show
+
+- "☰" to open the Show/Scene menu then Show tab
+- "+ New Show"
+- "Empty Show"
+- "Save" button, enter "my-baaahs", then "save"
+
+### Define a default projection
+
+- "☰" to open the Show/Scene menu then Show tab
+- Toggle "Design Mode"
+- "✎" next to the show name
+- New Projection Shader
```

Comment:
Maybe we just put this one into the shader library?

Reply:
Makes sense to me! Tried a quick spike on this but might have broken things. I'll follow up on moving this out of docs and into the library in a separate PR.

```diff
diff --git a/src/commonMain/resources/shader-libraries/built-in/_libraryIndex.json b/src/commonMain/resources/shader-libraries/built-in/_libraryIndex.json
index 85d9dbaaa..ecd8da297 100644
--- a/src/commonMain/resources/shader-libraries/built-in/_libraryIndex.json
+++ b/src/commonMain/resources/shader-libraries/built-in/_libraryIndex.json
@@ -3,6 +3,17 @@
   "description": "Some built-in shaders!",
   "license": null,
   "entries": [
+    {
+      "id": "flatprojection",
+      "title": "Flat Projection",
+      "description": "Project pixels onto a model",
+      "lastModifiedMs": 0,
+      "tags": [
+        "@filter",
+        "@type=projection"
+      ],
+      "srcFile": "projections/Flat.glsl"
+    },
     {
       "id": "beatshift",
       "title": "Beat Shift",
diff --git a/src/commonMain/resources/shader-libraries/built-in/projections/Flat.glsl b/src/commonMain/resources/shader-libraries/built-in/projections/Flat.glsl
new file mode 100644
index 000000000..c372856b2
--- /dev/null
+++ b/src/commonMain/resources/shader-libraries/built-in/projections/Flat.glsl
@@ -0,0 +1,17 @@
+// Flat Projection
+
+// Use this to project pixels onto a model
+
+struct ModelInfo {
+    vec3 center;
+    vec3 extents;
+};
+uniform ModelInfo modelInfo;
+
+// @return uv-coordinate
+// @param pixelLocation xyz-coordinate
+vec2 main(vec3 pixelLocation) {
+    vec3 start = modelInfo.center - modelInfo.extents / 2.;
+    vec3 rel = (pixelLocation - start) / modelInfo.extents;
+    return rel.xy;
+}
```

```
Missing incoming links on shader "Flat Projection
No link for ports: modelInfo, pixelLocation
```

Reply:
oh maybe I didn't have the a scene loaded

Reply:
@xian ok moved this to the library! LMK if I did it correctly... Could also split this out into its own PR if we want to keep this docs only.